go/vt/sqlparser/parse_test.go

```diff
@@ -610,6 +610,8 @@ func TestValid(t *testing.T) {
 		input: "select /* GE true */ 1 from t where a >= true",
 	}, {
 		input: "select /* GE false */ 1 from t where a >= false",
+	}, {
+		input: "select * from t order by a collate utf8_general_ci asc",
```

Comment:
it's up to @sougou but I feel like it's worth adding a test case for each of the forms the COLLATE keyword can be used: http://dev.mysql.com/doc/refman/5.7/en/charset-collate.html